101-azure-machine-learning-service-workspace/azuredeploy.json

```diff
@@ -0,0 +1,134 @@
+        "applicationInsightsName": "[concat('ai',uniqueString(resourceGroup().id))]",
+        "containerRegistryName": "[concat('cr',uniqueString(resourceGroup().id))]"
+    },
+    "resources": [
+        {
+            "name": "[variables('storageAccount').name]",
+            "type": "Microsoft.Storage/storageAccounts",
+            "location": "[parameters('workspaceLocation')]",
+            "apiVersion": "2018-07-01",
+            "sku": {
+                "name": "[variables('storageAccount').type]"
+            },
+            "kind": "StorageV2",
+            "properties": {
+                "encryption": {
+                    "services": {
+                        "blob": {
+                            "enabled": true
+                        },
+                        "file": {
+                            "enabled": true
+                        }
+                    },
+                    "keySource": "Microsoft.Storage"
+                },
+                "supportHttpsTrafficOnly": true
+            }
+        },
+        {
+            "name": "[variables('keyVault').name]",
+            "type": "Microsoft.KeyVault/vaults",
+            "apiVersion": "2018-02-14",
+            "location": "[parameters('workspaceLocation')]",
+            "properties": {
+                "tenantId": "[variables('keyVault').tenantId]",
+                "sku": {
+                    "name": "standard",
+                    "family": "A"
+                },
+                "accessPolicies": [],
+                "enabledForTemplateDeployment": true,
+                "enabledForDeployment": true,
+                "enabledForDiskEncryption": true
+            }
+        },
+        {
+            "name": "[variables('applicationInsightsName')]",
+            "type": "Microsoft.Insights/components",
+            "apiVersion": "2015-05-01",
+            "location": "[parameters('workspaceLocation')]",
+            "kind": "web",
+            "properties": {
+                "Application_Type": "web"
+            }
+        },
+        {
+            "name": "[variables('containerRegistryName')]",
+            "type": "Microsoft.ContainerRegistry/registries",
+            "apiVersion": "2017-10-01",
+            "location": "[parameters('workspaceLocation')]",
+            "sku": {
+                "name": "Standard"
+            },
+            "properties": {
+                "adminUserEnabled": true
+            }
+        },
+        {
+            "name": "[parameters('workspaceName')]",
+            "type": "Microsoft.MachineLearningServices/workspaces",
+            "apiVersion": "2018-11-19",
+            "location": "[parameters('workspaceLocation')]",
+            "dependsOn": [
+                "[variables('storageAccount').name]",
+                "[variables('keyVault').name]",
+                "[variables('applicationInsightsName')]",
+                "[variables('containerRegistryName')]"
+            ],
+            "identity": {
```

Comment:
how is this used?

Reply:
Assuming you are asking about "identity". I've pinged engineering to get info on what it does. It's not documented in the reference docs, and the template errors if it's removed. Waiting on response from engineering.

Reply:
The identity element uses a managed system identity (MSI). It's required when creating the workspace through the template. More info is available at https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/overview. From reading that, it sounds like an AD entry is created to represent the service when it is created, then removed when you remove the service.